e2e/specs/administration/predefinedResolutions.spec.ts

```diff
@@ -0,0 +1,182 @@
+import { logIn } from '../../pages/login.po';
+import { projectList } from '../../pages/project/list.po';
+import { notFound } from '../../pages/notFound.po';
+import { ProjectHelper } from '../../helpers/project.helper';
+import using from 'jasmine-data-provider';
+import usersTestData from '../../data/users.json';
+import { userAdministration } from '../../pages/administration/users.po';
+import { predefinedResolutions } from '../../pages/administration/predefinedResolutions.po';
+import resolutions from '../../data/resolutions.json';
+import cucumberImport from '../../data/import/cucumber.json';
+import { TestRun } from '../../../src/app/shared/models/testRun';
+import { TestResult } from '../../../src/app/shared/models/test-result';
+
+const editorExamples = {
+    admin: usersTestData.admin,
+    localAdmin: usersTestData.localAdmin,
+    localManager: usersTestData.localManager,
+    manager: usersTestData.manager,
+    localEngineer: usersTestData.localEngineer
+};
+
+const notEditorExamples = {
+    viewer: usersTestData.viewer,
+};
+
+const create = {
+    resolution: resolutions.global.testIssue,
+    comment: 'Test Issue',
+    expression: '.* Some error .*'
+};
+
+const edit = {
+    resolution: resolutions.global.appIssue,
+    comment: 'App Issue',
+    expression: '.*step.* was skipped.*'
+};
+
+const creationError = 'Fill all required fields';
+
+describe('Predefined Resolution:', () => {
+    const projectHelper: ProjectHelper = new ProjectHelper();
+
+    beforeAll(async () => {
+        await projectHelper.init({
+            localAdmin: usersTestData.localAdmin,
+            localManager: usersTestData.localManager,
+            localEngineer: usersTestData.localEngineer,
+            viewer: usersTestData.viewer,
+        });
+    });
+
+    afterAll(async () => {
+        await projectHelper.dispose();
+    });
+
+    using(editorExamples, (user, description) => {
+        describe(`${description} role:`, () => {
+            beforeAll(async () => {
+                await logIn.logInAs(user.user_name, user.password);
+                await projectHelper.openProject();
+            });
+
+            it('I can open Predefined Resolution page', async () => {
+                await (await projectList.menuBar.user()).administration();
+                await userAdministration.sidebar.predefinedResolutions();
+                return expect(predefinedResolutions.isOpened()).toBe(true, `Predefined Resolution page is not opened for ${description}`);
+            });
+
+            it('I can not create Predefined Resolution without Resolution', async () => {
```

Comment:
As I understand  name of scenario is not absolutely correct, because you don't click create button, so there is no attemt to create Predefined Resolution. I think test should click this button and check that error appears and resolution is not created

Reply:
Now it will check if creation button disabled